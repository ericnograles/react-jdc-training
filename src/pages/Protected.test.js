import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Protected from './Protected';

describe('<Protected>', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    let actions = {
      logout: jest.fn()
    };
    ReactDOM.render(<Protected actions={actions} />, div);
  });

  it('properly shows given text', () => {
    let actions = {
      logout: jest.fn()
    };
    const output = shallow(<Protected actions={actions} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should log out', () => {
    let actions = {
      logout: jest.fn()
    };
    const output = mount(<Protected actions={actions} />);
    output.find('button').simulate('click');
    expect(actions.logout).toHaveBeenCalled();
  });
});
