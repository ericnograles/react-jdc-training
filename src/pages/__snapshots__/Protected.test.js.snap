// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Protected> properly shows given text 1`] = `
<div>
  <p
    className="App-intro"
  >
    <button
      onClick={[Function]}
    >
      Logout
    </button>
  </p>
</div>
`;
